src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -146,7 +148,7 @@ public static IResourceBuilder<AzureSqlDatabaseResource> AddDatabase(this IResou
     /// builder.Build().Run();
     /// </code>
     /// </example>
-    public static IResourceBuilder<AzureSqlServerResource> RunAsContainer(this IResourceBuilder<AzureSqlServerResource> builder, Action<IResourceBuilder<SqlServerServerResource>>? configureContainer = null)
+    public static IResourceBuilder<AzureSqlServerResource> RunAsContainer(this IResourceBuilder<AzureSqlServerResource> builder, Action<IResourceBuilder<SqlServerServerResource>>? configureContainer = null, IResourceBuilder<ParameterResource>? password = null, int? port = null)
```

Comment:
@davidfowl @sebastienros @ReubenBond @JamesNK @DamianEdwards - are we concerned with binary breaking changes like this in the Hosting integrations? Part of me thinks this is OK, but wanted to check what everyone else thinks.

If we didn't go with this approach, we could take one of 2 approaches:

1. Just add 1 new overload that took all parameters, and callers would pass in `null` explicitly for the parameters they aren't supplying.
2. Explode the API out with 8 overloads - every combination of parameters - which feels excessive.

Reply:
An alternative is to expose some methods that allow for configuring the port and the password during the `configureContainer` callback.

See https://github.com/dotnet/aspire/commit/15e8a26aa14e6814bd02fa02fdb9d7811e639964 for an example:

```C#
var sql1 = builder.AddAzureSqlServer("sql1")
                  .RunAsContainer(c =>
                  {
                      c.WithHostPort(5555);
                  });
```

The downside with just exposing it on these 3 containers is that it isn't consistent with the rest of the "top level" containers - like MongoDB, RabbitMQ, etc. If we go with this route, would we add it to all of them as well?

Reply:
I think we've generally been fine with adding new optional parameters to existing methods right?

Reply:
We haven't been fine in the client integration libraries, since it is a binary breaking change. But I'm not sure we care as much in the hosting integrations libraries.

Reply:
Yeah I guess I'm basing this on what we've done in ASP.NET Core itself and `Aspire.Hosting`, or I'm totally misremembering 😄

Reply:
Don't make binary breaking changes in minors. Just add overloads 😄

Reply:
I think I'm in favor of taking the alternative approach above:

```C#
var sql1 = builder.AddAzureSqlServer("sql1")
                  .RunAsContainer(c =>
                  {
                      c.WithHostPort(5555);
                  });
```

This fits with our other "emulator" containers, and allows us to not have an explosion of overloads.

We would also then have a `.WithPassword(IResourceBuilder<ParameterResource>)` extension as well that allows you to set the password after the constructor.

Opinions on taking this approach?

Reply:
I thought about that as well but it would require more changes to the way that the underlying SqlServerResource is created/updated.

Reply:
I agree it would require some more changes throughout, but I think it provides a better API experience.